File: B's/cpp/2123B.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int t; cin >> t;
    vector<string> results;

    while (t--) {
        int n, j, k; 
        cin >> n >> j >> k;

        vector<int> s;

        while (n--) {
            int a; cin >> a;
            s.push_back(a);
        }

        if (k >= 2) {
            // If we're keeping at least 2 players, ANYONE can sneak into the last k.
            results.push_back("YES");
        } else {  // k == 1
            // Only the strongest can possibly be the sole survivor.
            int mx = *max_element(s.begin(), s.end());
            // j is 1-indexed
            if (s[j - 1] == mx) {
                results.push_back("YES");
            } else {
                results.push_back("NO");
            }
        }
    }

    for (int i = 0; i < results.size(); i++) {
        cout << results.at(i) << endl;
    }
    
    return 0;
}

File: B's/py/2123B.py
def main() -> None:
    t: int = int(input())
    res: list[str] = []

    while t > 0:
        n, j, k = map(int, input().split())
        s: list[int] = list(map(int, input().split()))

        if k >= 2:
            # If we're keeping at least 2 players, ANYONE can sneak into the last k.
            res.append("YES")
        else:  # k == 1
            # Only the strongest can possibly be the sole survivor.
            mx: int = max(s)
            # j is 1-indexed
            if s[j - 1] == mx:
                res.append("YES")
            else:
                res.append("NO")

        t -= 1

    for r in res: print(r)

if __name__ == "__main__":
    main()